unity-renderer/Assets/Desktop/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorDesktopFactory/ServiceLocatorDesktopFactory.cs

```diff
@@ -10,7 +12,7 @@ public static ServiceLocator CreateDefault()
             result.Register<IMemoryManager>(() => new MemoryManagerDesktop());
 
             // HUD
-            result.Register<IHUDFactory>(() => new HUDDesktopFactory());
+            result.Register<IHUDFactory>(() => new HUDDesktopFactory(result.Get<IAddressableResourceProvider>()));
```

Comment:
It's against the design to resolve dependencies during the registration. In DI it is fully forbidden, here you have a risk to use an uninitialized instance.
Just replace it with `Service<T>` in `HUDDesktopFactory`